test/integration/sequelize/query.test.js

```diff
@@ -266,22 +266,28 @@ describe(Support.getTestDialectTeaser('Sequelize'), () => {
         })).to.include('john');
       });
     } else if (dialect === 'db2') {
-      it('executes stored procedures', function () {
-        const self = this;
-
-        return self.sequelize.query(this.insertQuery).then(() => {
-          return self.sequelize.query('DROP PROCEDURE foo').then(() => {
-            return self.sequelize.query(
-              `CREATE PROCEDURE foo() DYNAMIC RESULT SETS 1 LANGUAGE SQL BEGIN DECLARE cr1 CURSOR WITH RETURN FOR SELECT * FROM ${qq(self.User.tableName)}; OPEN cr1; END`,
-            ).then(() => {
-              return self.sequelize.query('CALL foo()').then(users => {
-                expect(users.map(u => {
-                  return u.username;
-                })).to.include('john');
-              });
-            });
-          });
-        });
+      it('executes stored procedures', async function () {
+        const { sequelize } = this;
+
+        await sequelize.query(this.insertQuery);
+
+        try {
+          await sequelize.query('DROP PROCEDURE foo');
+        } catch (error) {
+          // DB2 does not support DROP PROCEDURE IF EXISTS
+          // -204 means "FOO" does not exist
+          // https://www.ibm.com/docs/en/db2-for-zos/11?topic=sec-204
+          if (error.cause.sqlcode !== -204) {
```

Comment:
Should we do assert(error.cause.sqlcode===204) ?

Reply:
No because the goal is to mute sql error -204 to emulate `DROP IF EXISTS` (as it means there is nothing to drop), but other errors should still surface and asserting here would shadow them